src/App.js

```diff
@@ -1,26 +1,127 @@
 import React from 'react';
-import logo from './logo.svg';
+import FoodBox from './components/FoodBox'
+import AddFood from './components/AddFood'
+import foods from './foods.json';
+import FoodList from './components/FoodList'
 import './App.css';
+import 'bulma/css/bulma.css';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+
+class App extends React.Component {
+  constructor(props){
+    super(props)
+    this.state = {
+      foods: foods,
+      filteredFoods: foods,
+      formShow: false,
+      todayFoods: []
+    }
+  }
+  
+
+  handleAdd = (event) => {
+    event.preventDefault()
+    let foodName = event.currentTarget.foodName.value
+    let foodCalories  = event.currentTarget.foodCalories.value
+    let foodImage = event.currentTarget.foodImage.value
+
+    let newFood = [
+      {
+        name: foodName,
+        calories: foodCalories,
+        image: foodImage,
+      }, ...this.state.foods]
+
+    this.setState({
+      foods: newFood,
+      formShow: false
+    })
+  }
+
+  handleShowForm = () => {
+    this.setState({
+      formShow: true
+    })
+  }
+
+  handleSearch = (event) => {
+    let searchName = event.currentTarget.value
+    let cloneFoods = this.state.foods.filter((item) => {
+      return item.name.startsWith(searchName)
```

Comment:
instead of startsWith you can use .includes. That one will do a better search functionality. Also convert both strings to either upperCase or lowerCase so it is not case sensitive. Check solution sent.